lib/kitchen/driver/ec2.rb

```diff
@@ -194,84 +247,176 @@ def default_aws_session_token
         env
       end
 
+      # First we check the existence of the metadata host.  Only fetch_credentials
+      # if we can find the host.
+      def iam_creds
+        require 'net/http'
+        require 'timeout'
+        @iam_creds ||= begin
+          timeout(5) do
+            Net::HTTP.get(URI.parse('http://169.254.169.254'))
+          end
+          fetch_credentials
+        rescue Errno::EHOSTUNREACH, Errno::EHOSTDOWN, Timeout::Error,
+          NoMethodError, ::StandardError => e
+          debug("fetch_credentials failed with exception #{e.message}:#{e.backtrace.join("\n")}")
+          {}
+        end
+      end
+
+      INSTANCE_METADATA_HOST = 'http://169.254.169.254'
+      INSTANCE_METADATA_PATH = '/latest/meta-data/iam/security-credentials/'
+      # fetch_credentials logic copied from Fog
+      def fetch_credentials
+        begin
+          connection = Excon.new(INSTANCE_METADATA_HOST)
```

Comment:
Might be able to get away with vanilla `net/http` which could trim excon as a gem dependency, but we could also look at that later in another release